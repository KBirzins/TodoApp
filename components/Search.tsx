import React, {useState} from 'react';
import styled from 'styled-components/native';

import FilterItem from './FilterItem';

const StyledView = styled.View`
  padding-top: 22px;
  padding-horizontal: 30px;
`;

const StyledHeaderText = styled.Text`
  text-align: center;
  font-size: 20px;
`;

const StyledTextInput = styled.TextInput`
  margin-left: 5px;
`;

const Search = () => {
  const [searchText, setText] = useState('');
  const [doneFilter, setDoneFilter] = useState(false);
  const [isDoneType, setDoneType] = useState(true);
  const [dateFilter, setDateFilter] = useState(false);
  const [dateType, setDateType] = useState(true);
  const [alphabetFilter, setAlphabetFilter] = useState(false);
  const [alphabetType, setAlphabetType] = useState(true);
  return (
    <StyledView>
      <StyledHeaderText>Search & Filter</StyledHeaderText>
      <FilterItem
        primaryText="Done"
        secondaryText="To do"
        isFilterActive={doneFilter}
        setFilterActive={setDoneFilter}
        filterType={isDoneType}
        setFilterType={setDoneType}
      />
      <FilterItem
        primaryText="Date Ascending"
        secondaryText="Date Descending"
        isFilterActive={dateFilter}
        setFilterActive={setDateFilter}
        filterType={dateType}
        setFilterType={setDateType}
      />
      <FilterItem
        primaryText="A-Z"
        secondaryText="Z-A"
        isFilterActive={alphabetFilter}
        setFilterActive={setAlphabetFilter}
        filterType={alphabetType}
        setFilterType={setAlphabetType}
      />

      <StyledTextInput
        onChangeText={text => setText(text)}
        placeholder="Search here"
        value={searchText}
      />
    </StyledView>
  );
};

export default Search;
