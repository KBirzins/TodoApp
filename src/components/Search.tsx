import React, {useState} from 'react';
import styled from 'styled-components/native';
import {useDispatch} from 'react-redux';

import FilterItem from './FilterItem';
import filterActions from '../reducers/filter/actions';

const StyledView = styled.View`
  padding-top: 22px;
  padding-horizontal: 30px;
`;

const StyledHeaderText = styled.Text`
  text-align: center;
  font-size: 20px;
`;

const StyledTextInput = styled.TextInput`
  margin-left: 5px;
`;

const Search = () => {
  const [searchText, setText] = useState('');
  const [doneFilter, setDoneFilter] = useState(false);
  const [isDoneType, setDoneType] = useState(true);
  const [dateFilter, setDateFilter] = useState(false);
  const [dateType, setDateType] = useState(true);
  const [alphabetFilter, setAlphabetFilter] = useState(false);
  const [alphabetType, setAlphabetType] = useState(true);

  const dispatch = useDispatch();
  dispatch(
    filterActions.setFilter({
      todo: {
        isActive: doneFilter,
        isPrimaryActive: isDoneType,
      },
      date: {
        isActive: dateFilter,
        isPrimaryActive: dateType,
      },
      alphabet: {
        isActive: alphabetFilter,
        isPrimaryActive: alphabetType,
      },
      text: searchText,
    }),
  );

  return (
    <StyledView>
      <StyledHeaderText>Search & Filter</StyledHeaderText>
      <FilterItem
        primaryText="Done"
        secondaryText="To do"
        isFilterActive={doneFilter}
        setFilterActive={setDoneFilter}
        filterType={isDoneType}
        setFilterType={setDoneType}
      />
      <FilterItem
        primaryText="Date Descending"
        secondaryText="Date Ascending"
        isFilterActive={dateFilter}
        setFilterActive={setDateFilter}
        filterType={dateType}
        setFilterType={setDateType}
      />
      <FilterItem
        primaryText="A-Z"
        secondaryText="Z-A"
        isFilterActive={alphabetFilter}
        setFilterActive={setAlphabetFilter}
        filterType={alphabetType}
        setFilterType={setAlphabetType}
      />

      <StyledTextInput
        onChangeText={text => setText(text)}
        placeholder="Search here"
        value={searchText}
        autoCapitalize="none"
      />
    </StyledView>
  );
};

export default Search;
