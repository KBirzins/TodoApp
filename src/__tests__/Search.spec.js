import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import {ReduxProvider} from '../tests/helpers';
import Search from '../components/Search';

describe('Search', () => {
  it('renders correctly with correct properties', () => {
    const component = renderer.create(
      <ReduxProvider>
        <Search />
      </ReduxProvider>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
