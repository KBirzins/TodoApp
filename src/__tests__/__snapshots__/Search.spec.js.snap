// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly with correct properties 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 30,
        "paddingTop": 22,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 20,
          "textAlign": "center",
        },
      ]
    }
  >
    Search & Filter
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingVertical": 5,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "color": "#211f30",
            "fontSize": 25,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "marginLeft": 5,
          },
        ]
      }
    >
      Done
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingVertical": 5,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "color": "#211f30",
            "fontSize": 25,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "marginLeft": 5,
          },
        ]
      }
    >
      Date Descending
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingVertical": 5,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "color": "#211f30",
            "fontSize": 25,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "marginLeft": 5,
          },
        ]
      }
    >
      A-Z
    </Text>
  </View>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Search here"
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "marginLeft": 5,
        },
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
